test-it/karma/events/ControlEvents.Test.js

```diff
@@ -64,7 +71,12 @@ describe('ValueChanged event tests', function () {
         var viPathParser = vireoRunner.createVIPathParser(vireo, 'UpdateBooleanOnValueChangeEvent');
 
         setTimeout(function () {
```

Comment:
not sure how i missed these setTimeouts but they probably need to be eliminated, definitely look like they can cause race conditions

Reply:
The settimeouts don't need to be addressed as part of this review, seems to be used elsewhere unexpectedly as well

Reply:
I added a new test that follows the same pattern. Is there a better way to do this other than setTimeout? I could update my new test and the others to follow that.

Reply:
We have a pattern that allows you to run JS immediately after the viaCode is loaded but before clumps have executed and another one after all clumps have executed. I think the first is likely too early (especially longer term when registration will be done with emitted JSLI instructions) and the latter is too late.

We would need to come up with a new pattern that is reliable. I'm thinking it would be polling periodically for a boolean flag that the via code sets and executing some JS after that time.
Or even the via test executes the JS via the JSLI.

But yea we would need to think of a new pattern, I don't think there is any good precedent.